tfhe/src/shortint/public_key/compact.rs

```diff
@@ -87,6 +87,64 @@ impl CompactPublicKey {
         })
     }
 
+    /// Deconstruct a [`CompactPublicKey`] into its constituants.
+    pub fn into_raw_parts(
+        self,
+    ) -> (
+        LweCompactPublicKeyOwned<u64>,
+        ShortintParameterSet,
+        PBSOrder,
+    ) {
+        let Self {
+            key,
+            parameters,
+            pbs_order,
+        } = self;
+
+        (key, parameters, pbs_order)
+    }
+
+    /// Construct a [`CompactPublicKey`] from its constituants.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the constituants are not compatible with each others.
+    pub fn from_raw_parts(
+        key: LweCompactPublicKeyOwned<u64>,
+        parameters: ShortintParameterSet,
+        pbs_order: PBSOrder,
+    ) -> Self {
+        let expected_pbs_order: PBSOrder = parameters.encryption_key_choice().into();
+
```

Comment:
I guess we should also check that `ciphertext_modulus` match
I think this applies to other places too

Reply:
places where there is parameters it is likely yes